lib/services/protocol_service.ts

```diff
@@ -572,11 +572,30 @@ export class SNProtocolService extends PureService implements EncryptionDelegate
         continue;
       }
       try {
+        let keyForPayload = key;
+        const version = encryptedPayload!.version;
+        const payloadContentType = encryptedPayload!.content_type;
+        const keyContentType = key?.content_type;
+        /** We check if the key can decrypt the encrypted payload. */
+        if (
+          version === ProtocolVersion.V004 &&
+          keyContentType === ContentType.RootKey &&
+          payloadContentType !== ContentType.ItemsKey
+        ) {
+          keyForPayload = undefined;
+        }
         const decryptedPayload = await this.payloadByDecryptingPayload(
           encryptedPayload,
-          key
+          keyForPayload
         );
-        decryptedPayloads.push(decryptedPayload);
+        /**
+         * If waitingForKey, we must import the payload in order to decrypt it later.
+         */
+        if (decryptedPayload.waitingForKey) {
+          await this.modelManager!.importPayloads([decryptedPayload]);
```

Comment:
@mobitar if a `Payload` has `waitingForKey = true` it must be imported into the items record:
https://github.com/standardnotes/snjs/blob/a27979314775f7aa34d4817bee740ab3fa3948c5/lib/services/protocol_service.ts#L601
The above line won't return anything because such items do not belong to the items record yet.

Reply:
https://github.com/standardnotes/snjs/blob/a27979314775f7aa34d4817bee740ab3fa3948c5/lib/application.ts#L787

The above will import all _valid items_ (i.e: `ItemsKey`)

Reply:
And then, the `ItemsKey` observer will have such `ItemsKey`s and will call [decryptErroredItems()](https://github.com/standardnotes/snjs/blob/a27979314775f7aa34d4817bee740ab3fa3948c5/lib/services/protocol_service.ts#L600) to attempt to decrypt invalid items (i.e: items which are `waitingForkey = true`):

https://github.com/standardnotes/snjs/blob/a27979314775f7aa34d4817bee740ab3fa3948c5/lib/services/protocol_service.ts#L143